### PR TITLE
Archive Students API Endpoint

### PR DESCRIPTION
**Pull request by CollinWoo**
### Summary 

This pull request implements a backend API endpoint for archiving students.

- [x] Added API endpoint for archiving students
- [x] Made sure that students can become unarchived if they re-request to be matched in a survey

### Test Plan 
Testing was performed by using Insomnia and the Firestore emulator to make sure requests to the new endpoint behaved as intended. This included checking that requesting the endpoint causes the corresponding `archived` field in the student's groups to be set to `false` and the corresponding student to be deleted from the course, and making sure that sending a match request for an archived class from the student survey unarchives the student from the course.

Insomnia Test showing successful API call

Screenshot 2023-03-10 at 9 04 47 PM

New `archived` field in Firestore Database
Screenshot 2023-03-10 at 9 09 59 PM

### Breaking Changes 

- Added an `archived` field to the `groups` field in each student document

### CODE REVIEW

File: backend/functions/src/student/functions.ts [hunk 1]
```diff
@@ -57,18 +57,40 @@ async function removeStudentFromCourse(
...
+  //remove unmatched student
+  if (groupNumber === -1) {
+    const ref = courseRef.doc(courseId)
+
+    return ref
+      .update({
+        unmatched: admin.firestore.FieldValue.arrayRemove(email),
+      })
+      .catch((err) => {
+        console.log(err)
```
Comment (wizhaaa) [inline backend/functions/src/student/functions.ts]: Not sure if we want to have console logging outside of testing - instead use Firebase logging for errors.

**Review by wizhaaa — APPROVED** [review]
Good work, Colin! I think this effectively solves the problem that LSC wanted to solve. So far it has worked when I call the endpoint. 

I think one potential thing we could do is add an archived column to the CSV exports. This could be another task though as still not sure if LSC wants it or not - will ask them about it.

**Review by b3njamint — COMMENTED** [review]
Good job Colin! This PR looks good to me, but make sure to remove the console.log statements as Will mentioned. I particularly liked how informative your error statements are :)

**Review by mluo24 — APPROVED** [review]
Sorry, just saw you rerequested this! I tested it out myself and it still works as intended. Thanks again, Colin!

**Pull request merged**